src/main/java/liquibase/ext/cassandra/sqlgenerator/DeleteGeneratorCassandra.java

```diff
@@ -24,15 +38,70 @@ public boolean supports(DeleteStatement statement, Database database) {
 	public Sql[] generateSql(DeleteStatement statement, Database database, SqlGeneratorChain sqlGeneratorChain) {
 
 		if (statement.getWhere() == null) {
-
-			String sql = "TRUNCATE " + database
-					.escapeTableName(statement.getCatalogName(), statement.getSchemaName(), statement.getTableName());
-			return new Sql[] { new UnparsedSql(sql, getAffectedTable(statement)) };
+			if (isAwsKeyspacesCompatibilityModeEnabled()) {
+				Scope.getCurrentScope().getLog(DeleteGeneratorCassandra.class)
+						.fine("AWS Keyspaces compatibility mode enabled: using alternative queries to truncate " + statement.getTableName());
+				return buildDeleteStatements(statement, database, sqlGeneratorChain).toArray(new Sql[]{});
+			} else {
+				String sql = "TRUNCATE " + database.escapeTableName(statement.getCatalogName(),
+						statement.getSchemaName(), statement.getTableName());
+				return new Sql[] { new UnparsedSql(sql, getAffectedTable(statement)) };
+			}
 		} else {
-
 			return super.generateSql(statement, database, sqlGeneratorChain);
 		}
 
 	}
 
+	/**
+	 * Builds a list of DELETE statements to remove each row of a given table.
+	 *
+	 * @implNote Since AWS Keyspaces does not support TRUNCATE TABLE statements (see
+	 * <a href="https://docs.aws.amazon.com/keyspaces/latest/devguide/cassandra-apis.html#cassandra-api-support">
+	 * Cassandra APIs in AWS Keyspaces</a>), this method tries to build a DELETE statement for each row in the table
+	 * to truncate, when the AWS Keyspaces compatibility mode is enabled.
+	 * Here, we cannot simply drop then re-create the table because the delete generator is too generic, so we can't
+	 * easily know the structure of the table to re-create.
+	 *
+	 * @param statement 		The original DELETE statement not having any WHERE clause.
+	 * @param database  		The database where the queries will be executed.
+	 * @param sqlGeneratorChain The SQL generator chain.
+	 * @return The list of DELETE statements to execute to truncate the given table.
+	 * @throws UnexpectedLiquibaseException in case something goes wrong during the preparation of the statements.
+	 */
+	private List<Sql> buildDeleteStatements(final DeleteStatement statement, final Database database,
+											final SqlGeneratorChain sqlGeneratorChain) {
+		List<Sql> sqlStatements = new ArrayList<>();
+
+		String tableName = database.escapeTableName(statement.getCatalogName(),
+				statement.getSchemaName(), statement.getTableName());
+
+		// Get the primary key values of each row in the table.
+		Table table = (Table) getAffectedTable(statement);
+		PrimaryKey pk = table.getPrimaryKey();
+		String pkColumnNames = pk.getColumnNames();
```

Comment:
We may have a nullpointer here if the table has no primary keys (like Liquibase DBCL table).

Reply:
In Cassandra, all tables must have a primary key. According to the [CQL documentation](https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlCreateTable.html): 
> A table must have at least one PRIMARY KEY

Also, as you can see here, even the DBCL table has a primary key in Cassandra: https://github.com/liquibase/liquibase-cassandra/blob/7c41a2c20e5388ecaea5910c794837d929bd564b/src/main/java/liquibase/ext/cassandra/sqlgenerator/CreateDatabaseChangeLogLockTableGeneratorCassandra.java#L29

But, I can obviously add a check to avoid NPE.

Reply:
🤦 don't worry about it, I should have RTFM. Thanks!